### PR TITLE
update(engine): modify append_output format

### PR DESCRIPTION
**Pull request by LucaGuerra**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

/kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

/area engine

**What this PR does / why we need it**:

Having discussed this feature with @leogr and @Issif we came to the conclusion that it could be better to modify the format of `append_output` to something like this example:

```yaml
append_output:
  - match:
      source: syscall
      tags: 
        - bar
        - foo
      rule: rule A
    extra_output: "gparent=%proc.aname[2] ggparent=%proc.aname[3] gggparent=%proc.aname[4]"
    extra_fields:
      - ka.verb
      - home_directory: "${HOME}"
      - my_field: "this is event number %evt.num"
```

`tags`, `source`, `rule` are ANDed together and also when multiple tags are specified those are all ANDed together as well.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by leogr — DISMISSED** [review]

File: userspace/engine/rule_loader_compiler.cpp [hunk 1]
```diff
@@ -501,9 +502,18 @@ void rule_loader::compiler::compile_rule_infos(
 				continue;
 			}
 
-			if (extra.m_tag != "" && r.tags.count(extra.m_tag) == 0)
+			if (extra.m_tags.size() != 0)
```
Comment (FedeDP) [inline userspace/engine/rule_loader_compiler.cpp:505]: Exctract an helper method since the same check is also used below?

**Comment by FedeDP** [comment]
I think we need to update configuration.cpp json schema too!
/hold

**Comment by LucaGuerra** [comment]
@FedeDP damn I forgot 🤣 , will fix. thanks

**Comment by leogr** [comment]
> I think we need to update configuration.cpp json schema too! /hold

Good catch!

side topic: it would be nice to introduce some automatic check

**Comment by FedeDP** [comment]
> side topic: it would be nice to introduce some automatic check

There is an automatic check that loads the falco.yaml from the CMAKE_CURRENT_SOURCE_DIR; but in this case, the new option is not present by default in the Falco yaml thus the test passes fine.

**Review by leogr — APPROVED** [review]

**Review by FedeDP — APPROVED** [review]
/approve

**Pull request merged**